lib/THC/generic/THCTensorMath.cu

```diff
@@ -391,4 +391,42 @@ accreal THCTensor_(trace)(THCState *state, THCTensor *src_) {
   return trace;
 }
 
+#if defined(THC_REAL_IS_FLOAT) || defined(THC_REAL_IS_DOUBLE) || defined(THC_REAL_IS_HALF)
+
+void THCTensor_(linspace)(THCState *state, THCTensor *r_, real a, real b, long n) {
+  THCAssertSameGPU(THCTensor_(checkGPU)(state, 1, r_));
```

Comment:
Just make it contiguous and do the freeCopyTo back to the original form (or whatever the corresponding functions are called these days, @soumith can help you, it's been a while since I've done work in cutorch). It will still operate on non-contiguous tensors this way.

I just don't think the overhead of adding a new version of the pointwise operators is justified in this case.